.github/workflows/check_helm_updates.yaml

```
@@ -0,0 +1,178 @@
+                  if [ "$deps" != "[]" ] && [ "$deps" != "null" ]; then
+                  return 0  # has dependencies
+                  fi
+              fi
+              return 1  # no dependencies
+          }
+
+          if [ "${{ github.event_name }}" = "workflow_dispatch" ] && [ ! -z "${{ github.event.inputs.charts }}" ]; then
+              # Use manually specified charts
+              CHARTS_JSON=$(echo "${{ github.event.inputs.charts }}" | tr ',' '\n' | jq -R -s -c 'split("\n")[:-1]')
+          else
+              for chart_dir in charts/*/; do
+                  chart=$(basename "$chart_dir")
+                  if check_dependencies "$chart_dir"; then
+                      charts_with_deps+=("$chart")
+                  else
+                  echo "Skipping ${chart} - no dependencies found"
+                  fi
+              done
+          fi
+
+          # Convert to JSON array for matrix strategy
+          CHARTS_JSON=$(printf '%s\n' "${charts_with_deps[@]}" | jq -R -s -c 'split("\n")[:-1]')
+          echo "chart-names=$CHARTS_JSON" >> $GITHUB_OUTPUT
+
+      - name: Set update strategies
+        id: set-strategies
+        run: |
+          if [ "${{ github.event_name }}" = "workflow_dispatch" ]; then
+            # Use manually specified strategy
+            STRATEGIES_JSON="[\"${{ github.event.inputs.update-strategy }}\"]"
+          else
+            # Use default strategies for scheduled run
+            STRATEGIES_JSON='["minor", "major"]'
+          fi
+          echo "strategies=$STRATEGIES_JSON" >> $GITHUB_OUTPUT
+
+  update-dependencies:
+    needs: discover-charts
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        chart: ${{ fromJson(needs.discover-charts.outputs.chart-names) }}
+        update-strategy: ${{ fromJson(needs.discover-charts.outputs.update-strategies) }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: "Upgrade Helm chart dependencies"
+        id: deps-update
+        uses: camptocamp/helm-dependency-update-action@v0.4.1
+        with:
+          chart-path: "charts/${{ matrix.chart }}"
+          update-strategy: "${{ matrix.update-strategy }}"
+          excluded-dependencies: "${{ github.event.inputs.excluded-dependencies }}"
+          dry-run: "${{ github.event.inputs.dry-run || false }}"
+
+      - name: "Bump local chart version"
+        if: ${{ steps.deps-update.outputs.update-type != 'none' }}
+        id: bump-version
+        run: |
+          CHART_YAML="charts/${{ matrix.chart }}/Chart.yaml"
+
+          # Get current version
+          CURRENT_VERSION=$(yq e '.version' "$CHART_YAML")
+          # Split version into major, minor, patch
+          IFS='.' read -r major minor patch <<< "$CURRENT_VERSION"
+          
+          # Update Chart.yaml with new version
+          # We're just updating the patch for now
+          yq e -i ".version = \"${major}.${minor}.$((patch + 1))\"" "$CHART_YAML"
```

nitpick: maybe its better to not have nested double quotes just for ease?
```suggestion
          yq e -i '.version = \"${major}.${minor}.$((patch + 1))\"' "$CHART_YAML"
```

---

doesn't work unfortunately - I think the yq expression needs double quotes and nested double quotes - I've tried different combinations

---

Ah ok good to know then thank you!